apps/electionguard-bindings/src/lib.rs

```diff
@@ -0,0 +1,180 @@
+    /// election.
+    pub election_parameters:  ElectionParameters,
+    /// The guardian public keys, which are registered in the election contract.
+    pub guardian_public_keys: Vec<GuardianPublicKey>,
+}
+
+impl TryFrom<EncryptedBallotContext> for PreVotingData {
+    type Error = JsError;
+
+    fn try_from(value: EncryptedBallotContext) -> Result<Self, Self::Error> {
+        let joint_election_public_key = JointElectionPublicKey::compute(
+            &value.election_parameters,
+            value.guardian_public_keys.as_slice(),
+        )
+        .map_err(|e| {
+            JsError::new(&format!(
+                "Could not compute joint election public key: {}",
+                e
+            ))
+        })?;
+
+        let hashes = Hashes::compute(&value.election_parameters, &value.election_manifest)
+            .map_err(|e| {
+                JsError::new(&format!(
+                    "Could not compute hashes from election context: {}",
+                    e
+                ))
+            })?;
+
+        let hashes_ext = HashesExt::compute(
+            &value.election_parameters,
+            &hashes,
+            &joint_election_public_key,
+            value.guardian_public_keys.as_slice(),
+        );
+
+        let pre_voting_data = PreVotingData {
+            manifest: value.election_manifest.clone(),
+            parameters: value.election_parameters,
+            hashes,
+            hashes_ext,
+            public_key: joint_election_public_key,
+        };
+
+        Ok(pre_voting_data)
+    }
+}
+
+/// Wrapper around a vector of bool flags, representing a selection of
+/// candidates for a single election guard contest.
+#[derive(Debug, Serialize, Deserialize, Tsify)]
+#[tsify(from_wasm_abi)]
+pub struct SingleContestSelection(pub Vec<bool>);
+
+impl From<SingleContestSelection> for BTreeMap<ContestIndex, ContestSelection> {
+    fn from(value: SingleContestSelection) -> Self {
+        let mut map = Self::new();
+        // We only ever have one contest, so we unwrapping value created from 1u8.
+        let index = ContestIndex::from_one_based_index_const(1).unwrap();
+        let value = ContestSelection {
+            vote: value.0.into_iter().map(|v| v.into()).collect(),
+        };
+
+        map.insert(index, value);
+        map
+    }
+}
+
+/// Get an encrypted ballot from a selection of candidates. The value returned
+/// matches the ballot format expected by the election contract entrypoint for
+/// registering ballots.
+#[wasm_bindgen(js_name = "getEncryptedBallot")]
+pub fn get_encrypted_ballot(
+    selections: SingleContestSelection,
+    context: EncryptedBallotContext,
+    device_uuid: String,
+) -> Result<js_sys::Uint8Array, JsError> {
+    let pre_voting_data: PreVotingData = context.try_into()?;
+    let device = Device::new(&device_uuid, pre_voting_data);
+
+    let mut seed = [0u8; 128];
+    thread_rng().fill_bytes(&mut seed);
```

Comment:
The max size of the output array is 32. I'll assume this is fine (otherwise please complain!). We can increase this by enabling a feature on the `rand` crate, but is it necessary?